middleware/Makefile

```diff
@@ -0,0 +1,12 @@
+.DEFAULT_GOAL := aarch64
+REPO_ROOT=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))/..
+
+ci:
+	./scripts/ci.sh
+envinit:
```

Comment:
The `make envinit` fails when the Go environment is not set up in a probably very specific way. I am currently missing this information on how to configure it so that dependencies are installed and found inside this repository. 

```
$ make envinit
./scripts/go-get.sh v1.15.0 github.com/golangci/golangci-lint/cmd/golangci-lint 
# cd /home/stadicus/go/src/github.com/golangci/golangci-lint; git pull --ff-only
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

package github.com/golangci/golangci-lint/cmd/golangci-lint: exit status 1
Makefile:7: recipe for target 'envinit' failed
make: *** [envinit] Error 1
```

Reply:
> The make envinit fails when the Go environment is not set up in a probably very specific way.

Yeah, I'm aware of this limitation, and considered removing `make envinit` or making it more maintainable, but since I was aiming for the minimum diff in this PR relative to the current upstream `github.com/shiftdevices/base-middleware` repo (for easier review and branch maintenance, in case upstream commits are added) I chose not to pursue those change in the current PR.

Let me know if you disagree with this approach and would like me to in the current PR either a) remove `make envinit` or b) add better documentation / automation to make it easier to run.

Reply:
If we agree that there is a gap in either documentation or script handling, I'm perfectly fine with the current pull request. I propose to create an additional issue to keep track of the work that needs to be done in that regard.

Reply:
> If we agree that there is a gap in either documentation or script handling, I'm perfectly fine with the current pull request.

Yup, agreed. I'll add a `TODO(hkjn)` next to the `make envinit` definition for now.

> I propose to create an additional issue to keep track of the work that needs to be done in that regard.

Sure, if you'd like; I can just send a follow-up after this PR as well.

Reply:
cc @TheCharlatan for the `make envinit` issue I mentioned in chat